roles/rollback/tasks/main.yml

```diff
@@ -2,14 +2,45 @@
 
 - name: remove current path
   file:
-    dest={{ current_path }}
-    state=absent
+    dest: "{{ current_path }}"
+    state: absent
 
 - name: copy rollback repo to current path
-  command: cp -r {{ rollback_path }} {{ current_path }} creates={{ current_path }}
+  command: cp -r {{ rollback_path }} {{ current_path }}
+  args:
+    creates: "{{ current_path }}"
 
-- name: copy rollback db to live db
-  shell: gunzip -c {{ rollback_sql_path }} | psql -h {{ db_host }} -U {{ db_user }} {{ db }}
+- name: close active db connections
+  shell: psql -c "REVOKE CONNECT ON DATABASE {{ db }} FROM public; SELECT pg_terminate_backend(pid) FROM pg_stat_activity WHERE datname='{{ db }}' AND pid <> pg_backend_pid();"
```

Comment:
AFAIK pg_stat_activity is not enabled at all. I'd like to whether this works at all.

Reply:
I tested it, and it seems to work. I tried a rollback on a Vagrant box with active connections to the app. It failed before I added this, and now it succeeds.

Reply:
I think the pg_stat_activiity table does some minimal things by default like track process ids, and can be configured to record additional monitoring data?

Reply:
Must be that. I'll give it a look.

Reply:
Nice! `pg_stat_activity` seems to be enabled by default. 

```
openfoodnetwork=> SELECT * FROM pg_stat_activity;
(...)
datid            | 16386
datname          | openfoodnetwork
pid              | 32597
usesysid         | 16384
usename          | ofn_user
application_name | delayed_job
client_addr      | ::1
client_hostname  | 
client_port      | 38148
backend_start    | 2019-04-15 12:40:28.499155+00
xact_start       | 
query_start      | 2019-04-17 13:51:37.313753+00
state_change     | 2019-04-17 13:51:37.313997+00
waiting          | f
state            | idle
backend_xid      | 
backend_xmin     | 
query            | UPDATE "delayed_jobs" SET locked_at = '2019-04-17 13:51:37.311119', locked_by = 'delayed_job host:scw-ecce22 pid:3
2594' WHERE id IN (SELECT  id FROM "delayed_jobs"  WHERE ((run_at <= '2019-04-17 13:51:37.308423' AND (locked_at IS NULL OR locked_at
 < '2019-04-17 13:36:37.308501') OR locked_by = 'delayed_job host:scw-ecce22 pid:32594') AND failed_at IS NULL) ORDER BY priority ASC
, run_at ASC LIMIT 1 FOR UPDATE) RETURNING *
```